### PR TITLE
lsp: make the error message more detailed

### PR DESCRIPTION
**Pull request by h-michael**

### CODE REVIEW

File: runtime/lua/vim/lsp.lua [hunk 1]
```diff
@@ -138,7 +138,7 @@ local function validate_command(input)
       end
     end
   else
-    error("cmd type must be string or list.")
+    error(string.format("cmd type must be string or list, but given %q.", type(input)))
```
Comment (justinmk) [inline runtime/lua/vim/lsp.lua]: could remove this `else` block and use `vim.validate` at the beginning of `validate_command`. That will also ensure that the error message has a standard format.

**Comment by tjdevries** [comment]
@h-michael should we still do something with this?

**Comment by h-michael** [comment]
@tjdevries No, ready :)

**Comment by tjdevries** [comment]
Can you fix the conflict w/ master?

**Comment by h-michael** [comment]
@tjdevries I've resolved conflict and add additional test.

File: runtime/lua/vim/lsp.lua [hunk 2]
```diff
@@ -524,7 +524,7 @@ function lsp.start_client(config)
   function client.request(method, params, callback, bufnr)
     if not callback then
       callback = resolve_callback(method)
-        or error("not found: request callback for client "..client.name)
+        or error(string.format("not found: %q request callback for client %q.", method, client.name))
```
Comment (tjdevries) [inline runtime/lua/vim/lsp.lua:527]: What is `%q`? I couldn't find it on lua docs.

Comment (h-michael) [reply runtime/lua/vim/lsp.lua:527]: See https://www.lua.org/manual/5.1/manual.html#pdf-string.format :)

Comment (tjdevries) [reply runtime/lua/vim/lsp.lua:527]: Excellent. ty

**Pull request merged**